src/buttons/docs/LoadingManager.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export default class LoadingManager extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.func
+  }
+
+  state = {
+    isLoading: false
+  }
+
+  render() {
+    return this.props.children({
+      isLoading: this.state.isLoading,
+      setLoading: () => {
```

Comment:
You should really make this a method on the class and keep the JSX clean.

Reply:
Yes.